src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkAddressChange.Linux.cs

```diff
@@ -35,7 +35,7 @@ public class NetworkChange
                 {
                     if (s_addressChangedSubscribers == null)
                     {
-                        Debug.Assert(s_socket == 0);
+                        Debug.Assert(s_socket == 0, "s_socket != 0, but there are no subscribers to NetworkAddressChanged.");
                         return;
                     }
 
```

Comment:
I added an assertion into CreateSocket() that checks the current s_socket == 0 before creating a new one.